tests/test_opthelper.py

```diff
@@ -0,0 +1,137 @@
+#! optimization of water using optHelper and explicit loop.
+#  1. Showing how to use external gradient.
+#  2. optking still has a module level parameters and history,
+#       that could be eliminated, so not yet multi-object safe.
+#  3. Have not yet restarted from json or disk, but should be close to working.
+import psi4
+import optking
+
+def test_step_by_step():
+    h2o = psi4.geometry("""
+         O
+         H 1 1.0
+         H 1 1.0 2 104.5
+    """)
+
+    psi4.core.clean_options()
+    psi4_options = {
+      'diis': False,
+      'basis': 'sto-3g',
+      'e_convergence': 10,
+      'd_convergence': 10,
+      'scf_type': 'pk',
+    }
+    psi4.set_options(psi4_options)
+
+    opt = optking.optHelper('hf')
+    opt.build_coordinates()
+    
+    for step in range(30):
+        opt.energy_gradient_hessian()
+        opt.step()
+        conv = opt.testConvergence()
+        if conv == True:
+            print("Optimization SUCCESS:")
+            break
+    else:
+        print("Optimization FAILURE:\n")
+     
+    # print(opt.history.summary_string())
+    json_output = opt.close()
+
+    E = json_output['energies'][-1] #TEST
+
+    nucenergy = json_output['trajectory'][-1]['properties']['nuclear_repulsion_energy']
+    refnucenergy =   8.9064983474  #TEST
+    refenergy    = -74.9659011923  #TEST
+    assert psi4.compare_values(refnucenergy, nucenergy, 3, "Nuclear repulsion energy")
+    assert psi4.compare_values(refenergy, E, 6, "Reference energy")
+
+
+def test_lj_external_gradient():
+    h2o = psi4.geometry("""
+         O
+         H 1 1.0
+         H 1 1.0 2 104.5
+    """)
+    
+    psi4.core.clean_options()
+    psi4_options = {
+      'basis': 'sto-3g',
+      'g_convergence': 'gau_verytight',
+    }
+    psi4.set_options(psi4_options)
+    
+    opt = optking.optHelper('hf',comp_type='user')
+    opt.build_coordinates()
+    
+    for step in range(30):
+        # Compute one's own energy and gradient
+        E, gX = optking.lj_functions.calc_energy_and_gradient(opt.geom, 2.5, 0.01, True)
+        # Insert these values into the 'user' computer.
+        opt.computer.external_energy = E
+        opt.computer.external_gradient = gX
+
+        opt.energy_gradient_hessian()
+        opt.step()
+        conv = opt.testConvergence()
```

Comment:
Instead of having the user set the attributes of the computer, it might be nice to have them interact only with `optHelper`. Since optHelper _has a_ computer. let the user call something like `opt.set_externals(E, gX, hessian=None)` and let optking handle setting the attributes for the computer.
Then call `opt.step()`. 
Add a 1 line call to `opt.energy_gradient_hessian()` to `opt.step()` to fetch needed information as needed from `get_pes_info()`.

i.e. the above would be:
```
opt.set_externals(E, gX)
opt.step()
conv = opt.testConvergence()
```

Reply:
I agree about the 'computer' but not so much about the function call.  Assigning a property is clean for the user, and flexible to changes on the back end.  So I like "opt.external_energy = E".  That I believe is "pythonic".  (I did build in that if you set the gradient you must also set the energy,etc.)

As far as combining energy_gradient_hessian and step together, we can of course eliminate as many of these calls as we like.  After all, we started with only one.  I just think we are going to need flexibility to handle use cases (which is the point of optHelper).  For example, the user can access the projected and constrained forces AND the guessed or updated Hessian after energy_gradient_hessian, but before taking the step.